plugins/modules/virtualdirectory_info.ps1

```diff
@@ -0,0 +1,81 @@
+#!powershell
+
+# Copyright: (c) 2024, Hen Yaish <hyaish@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+# Define the argument specification
+$spec = @{
+    options = @{
+        name = @{ required = $true; type = "str" }
+        site = @{ required = $true; type = "str" }
+        application = @{ type = "str" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$name = $module.Params.name
+$site = $module.Params.site
+$application = $module.Params.application
+
+try {
+    # Ensure WebAdministration module is loaded
+    if ($null -eq (Get-Module "WebAdministration" -ErrorAction SilentlyContinue)) {
+        Import-Module WebAdministration
+    }
+
+    # Construct the directory path (same logic as virtualdirectory module)
+    $directory_path = if ($application) {
+        "IIS:\Sites\$($site)\$($application)\$($name)"
+    }
+    else {
+        "IIS:\Sites\$($site)\$($name)"
+    }
+}
+catch {
+    $module.FailJson("Failed to ensure WebAdministration module is loaded.", $_.Exception.Message, $_ )
```

Comment:
We want the exception message to be included in the first arg, the second arg is the ErrorRecord/Exception. Right now this is trying to call an overload that doesn't exist.

```suggestion
    $module.FailJson("Failed to ensure WebAdministration module is loaded: $_", $_)
```